### PR TITLE
Fixup test for INSTALL_FROM_SDIST

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Description
The previous test for `INSTALL_FROM_SDIST` always failed due to a bad boolean test.

I guess in bash, whatever non empty string is provided returns true. Therefore, you should explicitely test for truthyness.

```bash
(test) ✔ ~/git/sk/scikit-image [data-files-illuminants ↑·1|✚ 1…1] 
23:28 $ bash test.sh 
false
(test) 23:28 $ INSTALL_FROM_SDIST=0 bash test.sh 
true
(test) ✔ ~/git/sk/scikit-image [data-files-illuminants ↑·1|✚ 1…1] 
23:28 $ INSTALL_FROM_SDIST=1 bash test.sh 
true
(test) ✔ ~/git/sk/scikit-image [data-files-illuminants ↑·1|✚ 1…1] 
23:28 $ cat test.sh 
if [[ $INSTALL_FROM_SDIST ]]; then
    echo true
else
    echo false
fi

```

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Review by mkcor — APPROVED** [review]
Thanks for fixing this!

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -84,8 +84,9 @@ jobs:
         run: |
             set -ex
             source tools/github/before_install.sh
+            set -ex
```
Comment (grlee77) [inline .github/workflows/tests.yml:87]: do we need "set -ex" a second time?

**Review by grlee77 — APPROVED** [review]
Thanks @hmaarrfk. I had a question about duplicate "set -ex" calls. Is the second one just to make sure it is still set after sourcing the script on the previous line?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (hmaarrfk) [reply .github/workflows/tests.yml:87]: becuase the source'd script removes it.... it was tricky to debug should i add a comment?

Comment (grlee77) [reply .github/workflows/tests.yml:87]: Okay, that is what I suspected. Let's go ahead and get this in as is

**Comment by hmaarrfk** [comment]
@meeseeksdev backport to v0.18.x

**Pull request merged**